wger/manager/views/workout.py

```diff
@@ -208,6 +212,73 @@ def add(request):
     return HttpResponseRedirect(workout.get_absolute_url())
 
 
+@login_required
+def export_json(request, pk):
+    '''
+    Implements the export of a workout as a JSON file
+    '''
+
+    workouts = Workout.objects.filter(user=request.user, id=pk)
+    exported_info = []
+
+    workout_info = {}
+    for workout in workouts:
+        workout_info['id'] = workout.id
+        workout_info['creation_date'] = str(workout.creation_date)
+        workout_info['comment'] = workout.comment
+        trainings = Day.objects.filter(training=workout.id)
+
+        if not trainings:
+            messages.info(request, 'You cannot export a workout without training days.'
+                                   ' Add training day.')
+            return HttpResponseRedirect(workout.get_absolute_url())
+
+        for training in trainings:
+            workout_info['workout_description'] = training.description
+            sets = Set.objects.filter(exerciseday_id=training)
+
+            workout_sets = []
+            for s in sets:
+                workout_sets.append(s.sets)
+            workout_info['sets'] = str(workout_sets)
+            workout_info['days'] = str(training.day.all())
+            set_exercises = workout.canonical_representation['day_list'][0]['set_list']
+            set_list = []
+
+            for set_exercise in set_exercises:
+                set_info = {}
+                list_of_exercises = []
+                exercise_info = {}
+
+                for info in set_exercise['exercise_list']:
+                    exercise_info['exercise'] = info['obj'].name
+                    exercise_info['set_number'] = info['setting_list']
```

Comment:
@tonykimathi why a list called set_number?

Reply:
@parseendavid I agree that the naming could be different. I've updated it. See changes.